infrastructure/modules/aws/eks/eks-cluster.tf

```diff
@@ -1,78 +1,86 @@
   cluster_name    = var.name
   cluster_version = var.eks.cluster_version
 
   # VPC
-  subnets = var.vpc.private_subnet_ids
-  vpc_id  = var.vpc.id
+  subnet_ids = var.vpc.private_subnet_ids
+  vpc_id     = var.vpc.id
 
-  # KUBECONFIG
-  kubeconfig_name                      = "kubeconfig-${var.name}"
-  write_kubeconfig                     = true
-  kubeconfig_output_path               = local.kubeconfig_output_path
-  kubeconfig_file_permission           = "0600"
-  kubeconfig_api_version               = "client.authentication.k8s.io/v1beta1"
-  kubeconfig_aws_authenticator_command = "aws"
-  kubeconfig_aws_authenticator_command_args = [
-    "--profile",
-    var.profile,
-    "--region",
-    local.region,
-    "eks",
-    "get-token",
-    "--cluster-name",
-    var.name
-  ]
+  create_aws_auth_configmap = true
+  # Needed to add self managed node group configuration.
+  # => kubectl get cm aws-auth -n kube-system -o yaml
+  manage_aws_auth_configmap = true
 
   # Private cluster
-  cluster_endpoint_private_access                = var.eks.cluster_endpoint_private_access
-  cluster_create_endpoint_private_access_sg_rule = var.eks.cluster_endpoint_private_access
-  cluster_endpoint_private_access_cidrs          = var.eks.cluster_endpoint_private_access_cidrs
-  cluster_endpoint_private_access_sg             = var.eks.cluster_endpoint_private_access_sg
+  cluster_endpoint_private_access = var.eks.cluster_endpoint_private_access
 
   # Public cluster
   cluster_endpoint_public_access       = var.eks.cluster_endpoint_public_access
   cluster_endpoint_public_access_cidrs = var.eks.cluster_endpoint_public_access_cidrs
 
   # Cluster parameters
-  cluster_enabled_log_types     = ["api", "audit", "authenticator", "controllerManager", "scheduler"]
-  cluster_log_kms_key_id        = var.eks.encryption_keys.cluster_log_kms_key_id
-  cluster_log_retention_in_days = var.eks.cluster_log_retention_in_days
-  cluster_create_security_group = true
-  cluster_encryption_config = [
-    {
-      provider_key_arn = var.eks.encryption_keys.cluster_encryption_config
-      resources        = ["secrets"]
-    }
-  ]
+  cluster_enabled_log_types              = ["api", "audit", "authenticator", "controllerManager", "scheduler"]
+  cloudwatch_log_group_kms_key_id        = var.eks.encryption_keys.cluster_log_kms_key_id
+  cloudwatch_log_group_retention_in_days = var.eks.cluster_log_retention_in_days
+  create_cluster_security_group          = true
+  cluster_encryption_config = {
+    provider_key_arn = var.eks.encryption_keys.cluster_encryption_config
+    resources        = ["secrets"]
+  }
 
   # Tags
+  #TODO: Arnaud.L tags must be added back - pb with depends_on on module 'bug'
   tags         = local.tags
   cluster_tags = local.tags
 
   # IAM
-  map_roles = concat([
+  # used to allow other users to interact with our cluster
+  aws_auth_roles = concat([
     {
-      rolearn  = module.eks.worker_iam_role_arn
+      rolearn  = local.cluster_iam_role_arn
```

Comment:
It is the iam_role_arn of each self_managed_node_group

Reply:
ok removed because already done by aws eks moduls itself